ClassAttendance/src/StudentApi/Controllers/StudentsController.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Threading.Tasks;
+using BLL.Exceptions;
+using BLL.Interfaces;
+using BLL.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace StudentApi.Controllers
+{
+    [Route("[controller]")]
+    public class StudentsController : Controller
+    {
+        private readonly IStudentService _studentService;
+        private readonly ILogger<StudentsController> _logger;
+
+        public StudentsController(IStudentService studentService, ILogger<StudentsController> logger)
+        {
+            _studentService = studentService;
+            _logger = logger;
+        }
+
+        [HttpGet("Students")]
+        public IActionResult GetAll()
+        {
+            var result = _studentService.GetAll();
+            _logger.LogInformation("Fetched students");
+
+            return Ok(result);
+        }
+
+        [HttpGet("Student")]
+        public async Task<IActionResult> GetByIdAsync(int id)
+        {
+            try
+            {
+                var user = await _studentService.GetByIdAsync(id);
+                _logger.LogInformation("Searched for student");
+
+                return Ok(user);
+            }
+            catch (BusinessLogicException ex)
+            {
+                _logger.LogError(ex.Message);
+                return BadRequest(ex.Message);
+            }
+        }
+
+        [HttpPost("Add")]
```

Comment:
I guess it would be better to have "Student" instead of "Add".
HTTP operation type will tell us about the action.
The same for all controllers.

Reply:
So i can use same address for CRUD operations, but with different http verbs? I'd rather change this names to something like "AddStudent", "DeleteStudent"

Reply:
It's useless and not match REST principles.